src/components/administrator/EditModal.jsx

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import style from './EditModal.module.css';
+import { useNavigate, Link, useParams, useLocation } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { useState, useEffect } from 'react';
+import { AiFillCloseCircle } from 'react-icons/ai';
+import { addProduct, encodeImageFileAsURL, getProductDetail, getProducts, updateProduct } from '../../data/API';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { myAtom } from '../../data/atoms.js';
+import LoadingModal from '../loading/LoadingModal.jsx';
+
+const EditModal = () => {
+  const atom = useRecoilValue(myAtom);
+
+  const navigate = useNavigate();
+  const { id } = useParams();
+
+  const { isLoading, data } = useQuery([id], () => getProductDetail(id));
+  const editProduct = useMutation(([id, payload]) => updateProduct(id, payload), {
+    // 성공하면 닫고 데이터 refetch
+    onSuccess: () => {
+      navigate('/admin/products');
+      atom.myFn(); // refetch 함수
```

Comment:
```suggestion
      window.location.href = '/admin/products'
```
혹시 이렇게 하는건 좋은방식이 아닌건가요 ..? 저는 페이지 이동 후 refetch가 필요한 상황일 때 저렇게 해버렸는데..

Reply:
`window.location.href = '/admin/products'` 로 페이지를 새로 불러오게 하며 브라우저가 화면 내 컴포넌트들을 전부 랜더링을 하게되지만,
리액트가 `navigate` 하거나 `리패치`를 한 경우, 바뀐 `state` 만 쏙 뽑아서 랜더링 하기 때문이에요